.github/workflows/build-and-push-image.yml

```diff
@@ -10,27 +10,42 @@ on:
         type: environment
         description: "Choose an environment to deploy to"
         required: true
-        default: "dev"
+        
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
 
 env:
   DOCKER_IMAGE: a2bint-app
-  NODE_VERSION: 18.x
 
 jobs:
-  build-and-push-image:
+  set-env:
+    name: Determine environment
     runs-on: ubuntu-22.04
-    environment: ${{ github.event.inputs.environment || "dev" }}
+    outputs:
+      environment: ${{ steps.var.outputs.environment }}
+      branch: ${{ steps.var.outputs.branch }}
+      release: ${{ steps.var.outputs.release }}
     steps:
-      - uses: actions/checkout@v2
-        with:
-          ref: ${{ github.ref }}
-
-      - name: Set environment variables
+      - id: var
         run: |
           GIT_REF=${{ github.ref }}
           GIT_BRANCH=${GIT_REF##*/}
-          echo "BRANCH_TAG=$GIT_BRANCH" >> $GITHUB_ENV
-          echo "DOCKER_IMAGE_TAG=$GITHUB_SHA" >> $GITHUB_ENV
+          INPUT=${{ github.event.inputs.environment }}
+          ENVIRONMENT=${INPUT:-"dev"}
+          RELEASE=${ENVIRONMENT,,}-`date +%Y-%m-%d`.${{ github.run_number }}
```

Comment:
Out of interest what do the comma's do in this instance?

Reply:
In `bash`, `$VAR^^` forces uppercase, whereas `$VAR,,` forces lowercase. This means that "Dev" becomes "dev" and "PROD" becomes "prod"